src/view/compareChangesTreeDataProvider.ts

```diff
@@ -173,8 +174,29 @@ export class CompareChangesTreeProvider implements vscode.TreeDataProvider<TreeN
  * Provides file contents for documents with GITHUB_FILE_SCHEME (githubpr) scheme. Contents are fetched from GitHub based on
  * information in the document's query string.
  */
-class GitHubContentProvider {
-	constructor(public gitHubRepository: GitHubRepository) { }
+class GitHubContentProvider extends ReadonlyFileSystemProvider {
+	constructor(public gitHubRepository: GitHubRepository) {
+		super();
+	}
+
+	async readFile(uri: any): Promise<Uint8Array> {
+		const params = fromGitHubURI(uri);
+		if (!params || params.isEmpty) {
+			return new TextEncoder().encode('');
+		}
+
+		const { octokit, remote } = await this.gitHubRepository.ensure();
+		const fileContent = await octokit.repos.getContent({
+			owner: remote.owner,
+			repo: remote.repositoryName,
+			path: params.fileName,
+			ref: params.branch,
+		});
+
+		const contents = (fileContent.data as any).content ?? '';
+		const buff = buffer.Buffer.from(contents, (fileContent.data as any).encoding);
+		return buff;
+	}
 
 	async provideTextDocumentContent(uri: vscode.Uri, _token: vscode.CancellationToken): Promise<string> {
```

Comment:
Looks like there are still references to `GitHubContentProvider` that expect it to `provideTextDocumentContent`, but I think to solve the notebook issue we need to replace all of those with the file system provider, correct?